server.go

```diff
@@ -396,6 +397,95 @@ func (s *server) GetPayment(ctx context.Context, in *breez.GetPaymentRequest) (*
 	return &breez.GetPaymentReply{PaymentError: sendResponse.PaymentError}, nil
 }
 
+func (s *server) GetSwapPayment(ctx context.Context, in *breez.GetSwapPaymentRequest) (*breez.GetSwapPaymentReply, error) {
+	redisConn := redisPool.Get()
+	m, err := redis.StringMap(redisConn.Do("HGETALL", "input-address:"+in.Address))
+	if err != nil {
+		log.Println("GetPayment error:", err)
+		return nil, status.Errorf(codes.Internal, "failed to retrieve address information")
+	}
+
+	// Ensure we didn't pay this invoice before
+	preImage := m["payment:PaymentPreimage"]
+	if preImage != "" {
+		return nil, status.Errorf(codes.Internal, "payment already sent")
+	}
+
+	// Decode the the client's payment request
+	clientCtx := metadata.AppendToOutgoingContext(context.Background(), "macaroon", os.Getenv("LND_MACAROON_HEX"))
+	decodedPayReq, err := client.DecodePayReq(clientCtx, &lnrpc.PayReqString{PayReq: in.PaymentRequest})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "payment request is not valid")
+	}
+	maxAllowedDeposit, err := getMaxAllowedDeposit(decodedPayReq.Destination)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to calculate max allowed deposit amount")
+	}
+	if decodedPayReq.NumSatoshis > maxAllowedDeposit {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("payment request amount: %v is greater than max allowed: %v", decodedPayReq.NumSatoshis, maxAllowedDeposit))
+	}
+	log.Printf("paying node %v amt = %v, maxAllowed = %v", decodedPayReq.Destination, decodedPayReq.NumSatoshis, maxAllowedDeposit)
+
+	// Validate the amount in database is the same as in payment request
+	amt, err := strconv.ParseInt(m["tx:Amount"], 10, 64)
```

Comment:
instead of taking the amount info from redis, I think we should call unspendAmount on lightninglib for this address.
It will give all the aggregated amount on several transactions if exists instead only the last one.